tests/components/sleepiq/test_sensor.py

```diff
@@ -1,67 +1,44 @@
 """The tests for SleepIQ sensor platform."""
-import unittest
-
-import requests_mock
-
 import homeassistant.components.sleepiq.sensor as sleepiq
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
 from tests.async_mock import MagicMock
-from tests.common import get_test_home_assistant
 from tests.components.sleepiq.test_init import mock_responses
 
+CONFIG = {"username": "foo", "password": "bar"}
 
-class TestSleepIQSensorSetup(unittest.TestCase):
-    """Tests the SleepIQ Sensor platform."""
-
-    DEVICES = []
-
-    def add_entities(self, devices):
-        """Mock add devices."""
-        for device in devices:
-            self.DEVICES.append(device)
-
-    def setUp(self):
-        """Initialize values for this testcase class."""
-        self.hass = get_test_home_assistant()
-        self.username = "foo"
-        self.password = "bar"
-        self.config = {"username": self.username, "password": self.password}
-        self.DEVICES = []
-        self.addCleanup(self.tear_down_cleanup)
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+async def test_setup(hass, requests_mock):
+    """Test for successfully setting up the SleepIQ platform."""
+    mock_responses(requests_mock)
 
-    @requests_mock.Mocker()
-    def test_setup(self, mock):
-        """Test for successfully setting up the SleepIQ platform."""
-        mock_responses(mock)
+    assert await async_setup_component(hass, "sleepiq", {"sleepiq": CONFIG})
 
-        assert setup_component(self.hass, "sleepiq", {"sleepiq": self.config})
+    device_mock = MagicMock()
+    sleepiq.setup_platform(hass, CONFIG, device_mock, MagicMock())
+    devices = device_mock.call_args[0][0]
+    assert 2 == len(devices)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)